cmd/posix.go

```diff
@@ -916,7 +916,10 @@ func deleteFile(basePath, deletePath string) error {
 	}
 
 	// Recursively go down the next path and delete again.
-	return deleteFile(basePath, slashpath.Dir(deletePath))
+	// Errors for parent directories shouldn't trickle down.
+	deleteFile(basePath, slashpath.Dir(deletePath))
```

Comment:
We don't @krishnasrinivas os.Remove() should have removed the expected file or directory by the time it reaches here. This is only the other empty/non-empty directories. If they failed for some reason it is okay to leave them as is.. since it wouldn't hurt IMO. 

If we didn't even return error for os.Remove() then it would mean something.  The problem here is that there is no locking involved when the directories are being recursively deleted upwards, which is the reason why inherently deleteFile() is racy and has unexpected results when a concurrent deleteFile() is running. 

I would suggest that we can safely do this change in most cases since our expected nature was to delete an empty directory or file.   If we did so we hope that it is possible for us to delete the top level dirs if they are empty as well, if its not say someone created a file while it is being recursively traversed we are okay to not return an error since as a context we have never held a lock on that part of the namespace anyways.